dbms/src/Storages/DeltaMerge/DeltaMergeStore.cpp

```diff
@@ -1500,6 +1501,17 @@ void DeltaMergeStore::checkSegmentUpdate(const DMContextPtr & dm_context, const
     {
         if (should_foreground_flush)
         {
+            Stopwatch watch;
+            SCOPE_EXIT({
+                // We may be flushing in BG threads. Do not count them as write stalls.
+                if (thread_type == ThreadType::Write)
+                {
+                    // FIXME: We'd better count write stall duration for per-write, instead of per-call,
+                    //   in order to produce a meaningful value.
+                    GET_METRIC(tiflash_storage_write_stall_duration_seconds, type_flush).Observe(watch.elapsedSeconds());
```

Comment:
How about name it more specifically, i.e. `type_fg_flush`.

Reply:
Operations in background will not be counted as write stalls (for now), so may be `fg` is redundant?